src/main/java/org/kuali/kfs/coa/document/validation/impl/AccountGlobalRule.java

```diff
@@ -863,5 +962,1065 @@ private boolean isIncomeStreamAccountRequired(String fundGroupCode, String subFu
+        			if (ObjectUtils.isNotNull(accountGlobalDetail.getAccount().getSubFundGroup())) {	
+	        			if (!getSubFundGroupService().isForContractsAndGrants(accountGlobalDetail.getAccount().getSubFundGroup())) {
+	        	                // this is not a C&G fund group. So users should not fill in any fields in the C&G tab.
+	        	                result &= checkCGFieldNotFilledIn(newAccount, accountGlobalDetail.getAccount(), KFSPropertyConstants.ACCT_INDIRECT_COST_RCVY_TYPE_CD);
+	        	                result &= checkCGFieldNotFilledIn(newAccount, accountGlobalDetail.getAccount(), KFSPropertyConstants.FINANCIAL_ICR_SERIES_IDENTIFIER);
+	        					result &= checkICRCollectionDoesNotExistForNonCGSubFund(newAccount, accountGlobalDetail,accountGlobalDetail.getAccount().getSubFundGroupCode());
+	        				}
+	        			else{
+	        				//sub fund group is CG and ICR should be filled
+	        				result &= checkICRCollectionExistsForCGSubFund(newAccount, accountGlobalDetail,accountGlobalDetail.getAccount().getSubFundGroupCode());
+	        			}  
+	                	}    			
+        		}
+        	}
+        }
+        return result;
+    }
+    
+    protected boolean checkICRCollectionExistsForCGSubFund(CuAccountGlobal newAccount, String subFundGroupCode){
+    	 boolean success = true;
+
+         	success = !newAccount.getActiveIndirectCostRecoveryAccounts().isEmpty();
+
+         	// if no active ICR entries in Acct Global check at account global detail level;check that existing accts at detail level are not being inactivated on the acct global
+         	if(!success){
+         		if(ObjectUtils.isNotNull(newAccount.getAccountGlobalDetails()) && !newAccount.getAccountGlobalDetails().isEmpty()){
+         			for(AccountGlobalDetail accountGlobalDetail : newAccount.getAccountGlobalDetails()){
+         				accountGlobalDetail.refreshReferenceObject(KFSPropertyConstants.ACCOUNT);
+         				List<IndirectCostRecoveryAccount> activeICRList = new ArrayList<IndirectCostRecoveryAccount>();
+         				
+         				for(IndirectCostRecoveryAccount icrAccount : accountGlobalDetail.getAccount().getActiveIndirectCostRecoveryAccounts()){
+         					boolean foundInactivated = false;
+         					for(IndirectCostRecoveryAccountChange icrAccountChange : newAccount.getIndirectCostRecoveryAccounts()){
+         						if(!icrAccountChange.isActive()){
+         							if(icrAccount.getChartOfAccountsCode().equalsIgnoreCase(icrAccountChange.getIndirectCostRecoveryFinCoaCode()) && icrAccount.getAccountNumber().equalsIgnoreCase(icrAccountChange.getIndirectCostRecoveryAccountNumber())){
+         								foundInactivated = true;
+         								break;
+         							}
+         						}
+         					}
+         					if(!foundInactivated){
+         						activeICRList.add(icrAccount);
+         					}
+         				}
+         				success &= !activeICRList.isEmpty();
+         	            if (!success){
```

Comment:
A couple of these validation methods have the "if (!success)" blocks in an odd horizontal position. Please update the horizontal positioning of these blocks accordingly for readability.